src/Features/LanguageServer/Protocol/Workspaces/LspWorkspaceManager.cs

```diff
@@ -407,15 +406,27 @@ private static async ValueTask<bool> AreChecksumsEqualAsync(Document document, S
 
     #endregion
 
+    /// <summary>
+    /// Get the language id for the given URI either from what the LSP client told us.
+    /// </summary>
+    internal string? GetLanguageForUri(Uri uri)
+    {
+        if (_trackedDocuments.TryGetValue(uri, out var trackedDocument))
```

Comment:
do we want to tie the language id to a document being opened?  It is totally possible for LSP to ask about a closed document.

Reply:
Done.